swap/tests/testutils/mod.rs

```diff
@@ -118,20 +122,27 @@ impl TestContext {
             .await
             .unwrap();
 
-        tokio::spawn(async move { event_loop.run().await });
+        let join_handle = tokio::spawn(async move { event_loop.run().await });
 
-        swap
+        (swap, BobEventLoopJoinHandle(join_handle))
     }
 
-    pub async fn recover_alice_from_db(&mut self) -> alice::Swap {
-        let (swap, mut event_loop) = self.alice_params.builder().await.build().await.unwrap();
+    pub async fn recover_alice_from_db(
```

Comment:
```suggestion
    pub async fn stop_and_resume_alice_from_db(
```
1. We use `resume` terminology everywhere else, let's remove a synonym (`recover`)
2. Express the fact that we simulate stopping the software (abort event loop) as part of this function